npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -43,10 +45,32 @@ def fetch_files
         fetched_files += workspace_package_jsons
         fetched_files += lerna_packages
         fetched_files += path_dependencies(fetched_files)
+        instrument_package_manager_version
 
         fetched_files.uniq
       end
 
+      def instrument_package_manager_version
+        package_managers = {}
+
+        if [package_lock].any?
+          [package_lock].each do |lockfile|
+            versions = [package_managers["npm"]]
+            versions << Helpers.npm_version_numeric(lockfile.content)
+            package_managers["npm"] = versions.compact.min
+          end
+        end
+
+        package_managers["yarn"] = 1 if [yarn_lock].any?
+        package_managers["shrinkwrap"] = 1 if [shrinkwrap].any?
```

Comment:
Since these are single files we can check them directly. Adding them to an array for an `.any?` check adds in some unnecessary indirection and it'd be best to keep it as simple as possible for future readers.

```suggestion
        package_managers["yarn"] = 1 if yarn_lock
        package_managers["shrinkwrap"] = 1 if shrinkwrap
```